pkg/stanza/fileconsumer/internal/tracker/tracker.go

```diff
@@ -26,6 +31,7 @@ type Tracker interface {
 	EndPoll()
 	EndConsume() int
 	TotalReaders() int
+	EnableArchiving(persister operator.Persister)
```

Comment:
@djaglowski I've renamed this function. It sounds much better to me. Let me know how it looks to you.

Reply:
It's not the name that worries me really. The problem is that instantiating the tracker and then having to reconfigure it later. Maybe the tracker should be instantiated entirely in Start?

Reply:
Makes sense. Opened a new PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35098 to tackle this, as they're independent changes.

Reply:
@djaglowski I meant https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35492. Sorry 🥲

Reply:
I don't see them as unrelated changes. The reason I suggested instantiating the tracker in `Start` is so that the persister can be passed to it when it is constructed. I don't see any point to #35492 without the archive.

Reply:
Ah, ok. Let me merge both in this then and we can get it done. Thanks!